lib/internal/worker.js

```diff
@@ -506,6 +506,42 @@ class Worker extends EventEmitter {
       };
     });
   }
+
+  // TODO(theanarkh): add options, such as sample_interval, CpuProfilingMode
+  startCpuProfile(name) {
+    validateString(name, 'name');
+    const startTaker = this[kHandle]?.startCpuProfile(name);
+    return new Promise((resolve, reject) => {
+      if (!startTaker) return reject(new ERR_WORKER_NOT_RUNNING());
+      startTaker.ondone = (err) => {
+        if (err) {
+          return reject(err);
+        }
+        let promise = null;
+        const stop = () => {
+          if (promise) {
+            return promise;
+          }
+          const stopTaker = this[kHandle]?.stopCpuProfile(name);
+          return promise = new Promise((resolve, reject) => {
+            if (!stopTaker) return reject(new ERR_WORKER_NOT_RUNNING());
+            stopTaker.ondone = (status, profile) => {
+              if (err) {
+                return reject(err);
+              }
+              resolve(profile);
+            };
+          });
+        };
+        resolve({
+          stop,
+          async [SymbolAsyncDispose]() {
+            await stop();
+          },
```

Comment:
Wouldn't this just discard the profile then? What use is there in starting a profile only to discard the result? 🤔